### PR TITLE
sec(graph-split): secure request entries

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

Secure graph-split

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

See jira issue

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/views/graphs/graph-split.php [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <?php
+
 /*
-* Copyright 2005-2015 Centreon
+* Copyright 2005-2021 Centreon
 * Centreon is developped by : Julien Mathis and Romain Le Merlus under
```
Comment (sc979) [inline www/include/views/graphs/graph-split.php:5]:
```suggestion
* Centreon is developed by : Julien Mathis and Romain Le Merlus under
```

**Review by sc979 — DISMISSED** [review]

File: www/include/views/graphs/graph-split.php [hunk 2]
```diff
@@ -48,174 +49,169 @@
...
-    while ($row = $res->fetch()) {
-        $metrics[] = array(
-            'id' => $svc_id . '_' . $row['metric_id'],
-            'title' => $row['host_name'] . ' - ' . $row['service_description'] . ' : ' . $row['metric_name']
-        );
-    }
+
+/* Get list metrics */
+$query = 'SELECT m.metric_id, m.metric_name, i.host_name, i.service_description
+    FROM metrics m, index_data i
```
Comment (callapa) [inline www/include/views/graphs/graph-split.php:65]: Please use the JOIN syntax

File: www/include/views/graphs/graph-split.php [hunk 3]
```diff
@@ -48,174 +49,169 @@
...
-/* Get Period if is in url */
-$period_start = 'undefined';
-$period_end = 'undefined';
-if (isset($_REQUEST['start']) &&
-    is_numeric($_REQUEST['start'])
-) {
-    $period_start = $_REQUEST['start'];
+if (isset($_GET['start'])) {
+    $period_start = filter_var($_GET['start'], FILTER_VALIDATE_INT);
 }
```
Comment (sc979) [inline www/include/views/graphs/graph-split.php:84]: As, if $_GET['start'] is not provided in the request, you'll fail the check line 195
https://github.com/centreon/centreon/pull/9738/files#diff-86135526cd024f1a05fa4e102aebd3b448d025713936b844c4476cbe7543554cR195

Same TODO for end_date
```suggestion
if (isset($_GET['start'])) {
    $period_start = filter_var($_GET['start'], FILTER_VALIDATE_INT) ?? 'undefined';
} else {
    $period_start = 'undefined';
}
```

File: www/include/views/graphs/graph-split.php [hunk 4]
```diff
@@ -48,174 +49,170 @@
...
-) {
-    $period_end = $_REQUEST['end'];
+
+if (isset($_GET['end'])) {
+    $period_end = filter_var($_GET['end'], FILTER_VALIDATE_INT);
+}
+
+if ($period_start === false) {
+    $period_start = 'undefined';
+}
```
Comment (sc979) [inline www/include/views/graphs/graph-split.php:92]:
```suggestion
```

**Review by sc979 — CHANGES_REQUESTED** [review]

**Review by sc979 — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**